### PR TITLE
DEV-2867: Add useEditablePageBatch

### PR DESCRIPTION
**Pull request by nrh-cklimas**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

1. https://github.com/newsrevenuehub/rev-engine/pull/843
2. https://github.com/newsrevenuehub/rev-engine/pull/882 
3. https://github.com/newsrevenuehub/rev-engine/pull/938
4. https://github.com/newsrevenuehub/rev-engine/pull/955 (this PR)
5. 🏁 

#### What's this PR do?

- Adds a new hook, useEditablePageBatch. The idea of the hook is that it manages a local set of page changes that can either be committed (set in the shared context) or reset (discarded).
- Refactored the Setup and Style tabs in the page editor to use this hook. The Layout and Sidebar tabs don't have changes because they don't have the same Undo/Update buttons.
- Adds comprehensive Jest coverage of the Setup tab. I didn't do this for Style because we're going to get rid of the tab in the near future.
 
#### Why are we doing this? How does it help us?

Improves testability, moves data logic out of components into a hook where it can be tested easier.

#### How should this be manually tested? Please include detailed step-by-step instructions.

We'll do QA on the feature branch as with other stacked PRs.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

No.

#### Have automated unit tests been added? If not, why?

Yes.

#### How should this change be communicated to end users?

n/a

#### Are there any smells or added technical debt to note?

See note about why there aren't tests for the Style tab.

I didn't migrate PageSetup to TypeScript (yet) because it is using some JS components still.

#### Has this been documented? If so, where?

No.

#### What are the relevant tickets? Add a link to any relevant ones.

https://news-revenue-hub.atlassian.net/browse/DEV-2867

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

No.

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

No.

### CODE REVIEW

File: spa/.storybook/preview.js [hunk 1]
```diff
@@ -1,10 +1,9 @@
 import { withThemes } from '@react-theming/storybook-addon';
 import { ThemeProvider as MuiThemeProvider } from '@material-ui/core';
 import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
-import { ThemeProvider } from 'styled-components';
-import { Provider as AlertProvider } from 'react-alert';
 import { SnackbarProvider } from 'notistack';
-import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { Provider as AlertProvider } from 'react-alert';
+import { ThemeProvider } from 'styled-components';
```
Comment (nrh-cklimas) [inline spa/.storybook/preview.js:6]: This was a side effect of syncing the feature branch with main, I think...

File: spa/cypress/e2e/02-donationPageEdit.cy.js [hunk 2]
```diff
@@ -574,7 +574,7 @@ describe('Edit interface: Setup', () => {
     cy.intercept({ method: 'GET', pathname: getEndpoint(USER) }, { body: orgAdminWithContentFlag });
     cy.intercept({ method: 'GET', pathname: getEndpoint(LIST_STYLES) }, {});
     cy.intercept(
-      { method: 'GET', pathname: `${getEndpoint(DRAFT_PAGE_DETAIL)}**` },
+      { method: 'GET', pathname: `${getEndpoint(LIST_PAGES)}**` },
```
Comment (nrh-cklimas) [inline spa/cypress/e2e/02-donationPageEdit.cy.js:577]: Same here, updating a test on main to use the correct URL.

File: spa/src/hooks/useEditablePageBatch.test.tsx [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    });
+
+    it('does nothing if called again without any new changes made', () => {
+      const { result } = tree();
+
+      act(() => result.current.addBatchChange({ header_link: 'test' }));
+      act(() => result.current.commitBatch());
+      setPageChanges.mockClear();
+      act(() => result.current.commitBatch());
+      expect(setPageChanges).not.toBeCalled();
```
Comment (GuiMend) [inline spa/src/hooks/useEditablePageBatch.test.tsx:139]: Shouldn't we add a `expect(setPageChanges).toBeCalled();` before `setPageChanges.mockClear();`?

File: spa/src/components/pageEditor/editInterface/pageSetup/PageSetup.test.js [hunk 4]
```diff
@@ -1,47 +1,180 @@
...
+    </PageEditorContext.Provider>
   );
 }
 
-it('should disable thank you page URL input and have tooltip if not enabled by plan', () => {
-  tree({ plan: { custom_thank_you_page_enabled: false, label: 'free' } });
-  expect(screen.queryByLabelText('Thank You page link')).not.toBeInTheDocument();
-});
+describe('PageSetup', () => {
+  const useEditablePageBatchMock = useEditablePageBatch; // as jest.Mock;
```
[the remark this answers is not in the document]
Comment (nrh-cklimas) [reply spa/src/components/pageEditor/editInterface/pageSetup/PageSetup.test.js:30]: I left this in as a note for when we convert the file to TypeScript, but if you think it's confusing I can remove it.

File: spa/src/hooks/useEditablePageBatch.test.tsx (hunk 3 above)
Comment (nrh-cklimas) [reply spa/src/hooks/useEditablePageBatch.test.tsx]: Makes sense to me--added in 43d58ab817b4f64a13c9c5dbe28dc34a24515583

File: spa/src/components/pageEditor/editInterface/pageSetup/PageSetup.test.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (GuiMend) [reply spa/src/components/pageEditor/editInterface/pageSetup/PageSetup.test.js:30]: No problem 👍

**Review by GuiMend — APPROVED** [review]

**Pull request merged**